### PR TITLE
Render HTML task descriptions instead of raw markup (#56)

### LINKED ISSUE #56: Task Description shows HTML Markup
**Description**
If a task has a description (i.e. a bulletpoint list) the description is show as HTML markup instead of showing the actual rendered content.

### PR DESCRIPTION
**Pull request by marco308**
## Summary
- Closes #56. Vikunja's web UI stores task descriptions as HTML (its Tiptap editor outputs `<ul>`, `<p>`, `<strong>`, etc.), but mDone only ever rendered descriptions as Markdown — so tasks edited on the web showed raw markup in both the preview and the editor.
- New `RichTextRenderer.render(_:)` helper detects HTML vs Markdown and renders each appropriately. HTML is parsed with `NSAttributedString(data:options:)` plus a small CSS preamble for the system font; foreground/background colors are stripped so dark mode still works. Non-HTML strings fall through to the existing inline-Markdown path.
- `TaskDetailSheet` (iOS) and `MacTaskDetailView` (macOS) now use the helper and default to preview mode when a description is present, so users see formatted content first and can toggle to edit the raw source.
- macOS preview is wrapped in a `ScrollView` so long descriptions don't clip.

## Test plan
- [x] `xcodegen generate`
- [x] iOS build (`xcodebuild -scheme mDone -sdk iphonesimulator -destination 'platform=iOS Simulator,name=iPhone 17' build`)
- [x] macOS build (`xcodebuild -scheme mDone-macOS build`)
- [x] SwiftLint clean on changed files
- [ ] Manual: open a task created in Vikunja's web UI with a bulleted-list description on iOS and macOS — verify list renders, toggle to edit reveals raw HTML, save still roundtrips the original content
- [ ] Manual: open a task with a plain-text/Markdown description — verify Markdown still renders as before

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by marco308** [comment]
Screenshots demonstrating the fix using a task created via Vikunja's web UI with this HTML description:

```html
<p>Agenda for the <strong>Q2 planning</strong> meeting:</p>
<ul>
  <li>Review Q1 metrics</li>
  <li>Set top-line goals</li>
  <li>Align on <em>hiring plan</em></li>
  <li>Approve budget for <a href="https://example.com">new tooling</a></li>
</ul>
<p>Prep deck before Friday.</p>
```

### iOS (iPhone 17 simulator)

| Before — raw HTML markup (the bug) | After — rendered (new default) |
| --- | --- |
|  |  |

### macOS

| Before — raw HTML markup | After — rendered (new default) |
| --- | --- |
|  |  |

Bold, italic, bulleted list, and links all render correctly in preview mode. Toggling the eye/pencil icon still reveals the raw HTML source for editing.

_Images are hosted on the `screenshots/issue-56` branch — safe to delete after merge._

File: mDone/Views/Tasks/RichTextRendering.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+enum RichTextRenderer {
+    static func render(_ source: String) -> AttributedString {
+        if containsHTML(source), let rendered = renderHTML(source) {
+            return rendered
+        }
+        return renderMarkdown(source)
+    }
+
+    static func containsHTML(_ source: String) -> Bool {
+        guard let regex = try? NSRegularExpression(pattern: "<[a-zA-Z/][^>]*>") else { return false }
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/Views/Tasks/RichTextRendering.swift]: Good catch — fixed in c826748. Detection now requires a closing tag (`</...>`), so Markdown autolinks like `<https://example.com>` stay on the Markdown path. Covered by `testContainsHTMLIgnoresMarkdownAutolink` and `testRenderMarkdownAutolinkPreservesText`.

File: mDone/Views/Tasks/RichTextRendering.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        let range = NSRange(source.startIndex..<source.endIndex, in: source)
+        return regex.firstMatch(in: source, options: [], range: range) != nil
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/Views/Tasks/RichTextRendering.swift:24]: Done in c826748 — regex is now a `static let htmlClosingTagPattern`, compiled once.

File: mDone/Views/Tasks/RichTextRendering.swift [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import Foundation
+import SwiftUI
+
+#if canImport(UIKit)
+    import UIKit
+#elseif canImport(AppKit)
+    import AppKit
+#endif
+
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/Views/Tasks/RichTextRendering.swift]: Partially — `SwiftUI` removed in c826748. `UIKit`/`AppKit` are still needed: the HTML `NSAttributedString(data:options:documentAttributes:)` initializer is declared in those frameworks (it isn't in plain Foundation), and `UIFont`/`NSFont` are used by the new Dynamic Type rewrite.

File: mDone/Views/Tasks/RichTextRendering.swift [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    private static func renderHTML(_ html: String) -> AttributedString? {
+        let styled = """
+        <meta charset="utf-8">
+        <style>
+          body { font-family: -apple-system, BlinkMacSystemFont, "Helvetica Neue", sans-serif; font-size: 16px; line-height: 1.4; }
+          p, ul, ol { margin: 0 0 8px 0; }
+          ul, ol { padding-left: 1.4em; }
+          code { font-family: ui-monospace, Menlo, monospace; }
+        </style>
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/Views/Tasks/RichTextRendering.swift:43]: Fixed in c826748. Dropped `font-size` from the CSS preamble and added `rewriteFontsForDynamicType` that walks the parsed runs, extracts bold/italic traits, and reapplies them on top of `UIFont.preferredFont(forTextStyle: .body)` (or `NSFont` equivalent on macOS). Preview now scales with the user's text size while keeping inline bold/italic.

File: mDone/Views/Tasks/TaskDetailSheet.swift [hunk 5]
```diff
@@ -17,18 +17,20 @@ struct TaskDetailSheet: View {
...
         _title = State(initialValue: task.title)
-        _description = State(initialValue: task.description ?? "")
+        _description = State(initialValue: initialDescription)
         _dueDate = State(initialValue: task.effectiveDueDate)
         _hasDueDate = State(initialValue: task.effectiveDueDate != nil)
         _priority = State(initialValue: task.priority)
         _selectedProjectId = State(initialValue: task.projectId)
         _repeatInterval = State(initialValue: task.repeatAfter ?? 0)
         _reminders = State(initialValue: task.reminders ?? [])
+        _isPreviewingMarkdown = State(initialValue: !initialDescription.isEmpty)
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/Views/Tasks/TaskDetailSheet.swift]: Renamed to `isShowingDescriptionPreview` in c826748.

File: mDone/Views/Mac/MacTaskDetailView.swift [hunk 6]
```diff
@@ -13,18 +13,20 @@ struct MacTaskDetailView: View {
...
         _title = State(initialValue: task.title)
-        _descriptionText = State(initialValue: task.description ?? "")
+        _descriptionText = State(initialValue: initialDescription)
         _dueDate = State(initialValue: task.dueDate)
         _hasDueDate = State(initialValue: task.dueDate != nil)
         _priority = State(initialValue: task.priority)
         _selectedProjectId = State(initialValue: task.projectId)
         _repeatInterval = State(initialValue: task.repeatAfter ?? 0)
         _reminders = State(initialValue: task.reminders ?? [])
+        _isPreviewingMarkdown = State(initialValue: !initialDescription.isEmpty)
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/Views/Mac/MacTaskDetailView.swift]: Renamed to `isShowingDescriptionPreview` and updated the `.help(...)` text from "Preview Markdown" to "Preview description" in c826748.

File: mDone/Views/Tasks/RichTextRendering.swift [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    import AppKit
+#endif
+
+enum RichTextRenderer {
+    static func render(_ source: String) -> AttributedString {
+        if containsHTML(source), let rendered = renderHTML(source) {
+            return rendered
+        }
+        return renderMarkdown(source)
+    }
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/Views/Tasks/RichTextRendering.swift:15]: Added [`RichTextRendererTests.swift`](https://github.com/marco308/mdone/blob/claude/confident-liskov-da56a8/mDoneTests/RichTextRendererTests.swift) in c826748 — 10 tests covering HTML vs Markdown routing, the autolink edge case, plain-text round-trip, and list/strong rendering. All passing locally.
